### PR TITLE
MIDI-32: Investigate the difference between classification and distribution modelling

### PR DESCRIPTION
**Pull request by JasiekKaczmarczyk**
[Investigate the difference between classification and distribution modelling](https://sacrebleu.atlassian.net/browse/MIDI-32)

Implemented alternative training loop for training Unet for distribution modelling. Now model instead of predicting logits that are used for classification is directly modelling distribution of masks.

Here is result on test set after 1 epoch of training:
model_distribution_modelling_test_set

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,6 +23,55 @@ python train.py --config-name <config>
...
+  lr: 3e-4
+  weight_decay: 0.01
+  num_epochs: 1
+  device: "cuda"
+  precision: "16-mixed" # not implemented yet
+  overfit_batches: 0
+
+paths:
+  save_ckpt_dir: "checkpoints" # directory where checkpoints will be saved
+  load_ckpt_path: null # if not None, specifies path to model state dict which will be loaded
```
Comment (roszcz) [inline README.md]: I'd say there's no reason to store a full config in readme - I think it's better to  just show a command that's guaranteed to work, i.e.:

```sh
# This would be L22 of this README
python train.py --config-name config-distribution-modelling
```
And mention that there are different configs available.

File: tests/unet_classification_test.py [hunk 2]
```diff
@@ -1,18 +1,31 @@
...
+from train import preprocess_dataset
+from train_binary_classification import intersection_over_union
 
 if __name__ == "__main__":
     # initializing model
-    ckpt = torch.load("checkpoints/run-2023-08-14-09-24.ckpt")
-    model = Unet(in_channels=2, out_channels=1, dim=32, dim_mults=(1, 2, 4), kernel_size=7, resnet_block_groups=4)
-    model.load_state_dict(ckpt["model"])
+    checkpoint = torch.load(
+        hf_hub_download(repo_id="JasiekKaczmarczyk/ecg-segmentation-unet", filename="classification-2023-08-14-09-24.ckpt")
```
Comment (roszcz) [inline tests/unet_classification_test.py:14]: Can you also add code showing how to upload checkpoints to huggingface? :)

**Review by roszcz — APPROVED** [review]
Great stuff! :rocket: 

I have a small issue with the `tqdm` - progress bar looks like this right now:

```sh
99it [10:22,  6.71s/it, loss=0.00831]
```

Which doesn't tell me anything percentage wise about the progress. The way you're defining the `train/val_loop` right now, you can add the "total" parameter:

```python
train_loop = tqdm(enumerate(train_dataloader), total=len(train_dataloader))
#  41%|█████████████████                                       | 2366/5706 [03:53<04:49, 11.54it/s]
```

And tqdm will create actuall progress bar. There are other ways to display this information, but it's important for me to see how long is the epoch going to take

File: tests/unet_classification_test.py (hunk 2 above)
Comment (JasiekKaczmarczyk) [reply tests/unet_classification_test.py:14]: I added it into train.py at the end.
